src/idl_parser.cpp

```diff
@@ -428,11 +428,27 @@ CheckedError Parser::Next() {
           if (c == '0' && (*cursor_ == 'x' || *cursor_ == 'X')) {
             cursor_++;
             while (isxdigit(static_cast<unsigned char>(*cursor_))) cursor_++;
-            attribute_.append(start + 2, cursor_);
-            attribute_ = NumToString(static_cast<int64_t>(
+            if (*cursor_ != '.') // Otherwise it might be a hexfloat, so lets handle this scenario down below
+            {
```

Comment:
Please use the style of the rest of this file, e.g. `{` not on its own line. Use clang-format if needed.